if revision is empty oomcli join will panic

recurrent:

```shell
$ oomcli init
$ oomcli apply -f /tmp/tmp/meta.yaml
$ oomcli get meta entity -o yaml
kind: Entity
name: user
description: user
groups:
    - name: user-click
      category: stream
      snapshot-interval: 1s
      description: user click post feature
      features:
        - name: last_5_click_posts
          value-type: string
          description: user last 5 click posts
        - name: number_of_user_starred_posts
          value-type: int64
          description: number of posts that users stared today
$ cat /tmp/tmp/entity_rows.csv
entity_key,unix_milli
1,100
2,100
$ oomcli join --feature user-click.last_5_click_posts,user-click.number_of_user_starred_posts --input-file /tmp/tmp/entity_rows.csv --output csv                fish-0 | 0 (0.001s) < 10:37:28
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.DoJoin({0x1d624e8, 0xc0000400a0}, {{0x1a6f45a, 0x6}, 0xc0004d8270, 0x0, 0x0}, {{{0x1, {0xc00072d610, 0x4}, ...}, ...}, ...})
        /home/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join.go:81 +0xac5
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.Join({0x1d624e8, 0xc0000400a0}, 0x410f76, {{0x1, {0xc00072d610, 0x4}, {0xc00072d640, 0x4}, {0x0, 0xed980057a, ...}, ...}, ...}, ...
)
        /home/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join.go:44 +0x15f
github.com/oom-ai/oomstore/internal/database/offline/sqlite.(*DB).Join(0x80, {0x1d624e8, 0xc0000400a0}, {{0x1, {0xc00072d610, 0x4}, {0xc00072d640, 0x4}, {0x0, 0xed980057a, ...}, ...}, ...})
        /home/lianxm/github.com/oomstore/internal/database/offline/sqlite/store.go:45 +0x6d
github.com/oom-ai/oomstore/pkg/oomstore.(*OomStore).ChannelJoin(0xc00013ee40, {0x1d624e8, 0xc0000400a0}, {{0xc0003aec60, 0x2, 0x2}, {0xc0003af580, 0x0, 0x0}, 0xc00032ab40})
        /home/lianxm/github.com/oomstore/pkg/oomstore/join.go:64 +0x687
github.com/oom-ai/oomstore/oomcli/cmd.join({0x1d624e8, 0xc0000400a0}, 0xc0003f5cb0, {{0x7ffd4eb22325, 0x18}, {0xc0003aec60, 0x2, 0x2}}, {0x7ffd4eb22347, 0x3})
        /home/lianxm/github.com/oomstore/oomcli/cmd/join_helper.go:27 +0x158
github.com/oom-ai/oomstore/oomcli/cmd.glob..func24(0x2abb960, {0x1a6c160, 0x6, 0x6})
        /home/lianxm/github.com/oomstore/oomcli/cmd/join.go:25 +0x108
github.com/spf13/cobra.(*Command).execute(0x2abb960, {0xc0004b3a40, 0x6, 0x6})
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x2abc860)
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/oom-ai/oomstore/oomcli/cmd.Execute()
        /home/lianxm/github.com/oomstore/oomcli/cmd/root.go:32 +0x25
main.main()
        /home/lianxm/github.com/oomstore/oomcli/main.go:6 +0x17
```

--- a/internal/database/offline/sqlutil/join.go
+++ b/internal/database/offline/sqlutil/join.go
@@ -180,16 +180,15 @@ func JoinOneGroup(ctx context.Context, dbOpt dbutil.DBOpt, opt offline.JoinOneGr
 }
 
 func ReadJoinedTable(ctx context.Context, dbOpt dbutil.DBOpt, opt ReadJoinedTableOpt) (*types.JoinResult, error) {
-	data := make(chan []interface{})
-	defer close(data)
-	emptyResult := &types.JoinResult{
-		Data: data,
-	}
-	tableNames := opt.TableNames
-	dropTableNames := append([]string{opt.EntityRowsTableName}, opt.AllTableNames...)
-	if len(tableNames) == 0 {
+	if len(opt.TableNames) == 0 {
+		data := make(chan []interface{})
+		defer close(data)
+		emptyResult := &types.JoinResult{
+			Data: data,
+		}
 		return emptyResult, nil
 	}
+
 	qt := dbutil.QuoteFn(dbOpt.Backend)
 
 	// Step 1: join temporary tables
@@ -225,7 +224,7 @@ func ReadJoinedTable(ctx context.Context, dbOpt dbutil.DBOpt, opt ReadJoinedTabl
 			ValueType: types.String,
 		})
 	}
-	for _, name := range tableNames {
+	for _, name := range opt.TableNames {
 		for _, f := range opt.FeatureMap[name] {
 			fields = append(fields, fmt.Sprintf("%s.%s", qt(name), qt(f.Name)))
 			header = append(header, dbutil.Column{
@@ -235,7 +234,7 @@ func ReadJoinedTable(ctx context.Context, dbOpt dbutil.DBOpt, opt ReadJoinedTabl
 		}
 	}
 
-	tableNames = append([]string{opt.EntityRowsTableName}, tableNames...)
+	tableNames := append([]string{opt.EntityRowsTableName}, opt.TableNames...)
 	for i := 0; i < len(tableNames)-1; i++ {
 		joinTablePairs = append(joinTablePairs, joinTablePair{
 			LeftTable:  tableNames[0],
@@ -258,6 +257,8 @@ func ReadJoinedTable(ctx context.Context, dbOpt dbutil.DBOpt, opt ReadJoinedTabl
 	if err != nil {
 		return nil, err
 	}
+
+	dropTableNames := append([]string{opt.EntityRowsTableName}, opt.AllTableNames...)
 
 	// Step 2: read joined results
 	return opt.QueryResults(ctx, dbOpt, query, header, dropTableNames, dbOpt.Backend)

--- a/pkg/oomstore/join.go
+++ b/pkg/oomstore/join.go
@@ -58,6 +58,9 @@ func (s *OomStore) ChannelJoin(ctx context.Context, opt types.ChannelJoinOpt) (*
 		if err != nil {
 			return nil, err
 		}
+		if len(revisionRanges) == 0 {
+			return nil, errdefs.Errorf("group %s no feature values", groupName)
+		}
 		revisionRangeMap[groupName] = revisionRanges
 	}
 

